components/sst_importer/src/sst_importer.rs

```diff
@@ -522,16 +528,9 @@ impl<E: KvEngine> SSTWriter<E> {
         Ok(metas)
     }
 
-    fn save(
-        mut sst_reader: <<E as SstExt>::SstWriter as SstWriter>::ExternalSstFileReader,
-        import_path: ImportPath,
-    ) -> Result<()> {
-        let tmp_path = import_path.temp;
-        let mut tmp_f = File::create(&tmp_path)?;
-        std::io::copy(&mut sst_reader, &mut tmp_f)?;
-        tmp_f.metadata()?.permissions().set_readonly(true);
-        tmp_f.sync_all()?;
-        fs::rename(tmp_path, import_path.save)?;
+    // move file from temp to save.
+    fn save(import_path: ImportPath) -> Result<()> {
+        fs::rename(import_path.temp, import_path.save)?;
```

Comment:
Better to fsync the `import_path` directory.

Reply:
It's not easy to do a fsync to the `import_path` directory here, because import_path only contains path info for the 3 files. We need to do following to sync the dir:
```
import_path.save.pop();
file::sync_all(&import_path.save);
```
Not sure if this action is really necessary since the need to use pop to get directory is somewhat weird to me.  And the original code only sync the written file but don't sync the directory.

Reply:
It is not safe rename without fsync, see https://github.com/facebook/rocksdb/issues/5287

You can call `sync_dir`.
https://github.com/tikv/tikv/blob/7935019849d46b7d32b1a6b0d14e795cd7da1591/components/tikv_util/src/file.rs#L52